src/mlpack/core/cv/metrics/ssim_index_impl.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef MLPACK_CORE_CV_METRICS_SSIMINDEX_IMPL_HPP
+#define MLPACK_CORE_CV_METRICS_SSIMINDEX_IMPL_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace cv {
+
+template<typename DataType>
+double SSIMIndex::Evaluate(DataType const& image,
+                           DataType const& reference)
+{
+ if (arma::size(reference) != arma::size(image))
+ {
+  std::ostringstream oss;
+  oss << "SSIMIndex::Evaluate(): size of reference (" 
+    <<reference.n_cols<<"X"<<reference.n_rows
+    <<") does not match size of image ("
+    <<image.n_cols<<"X"<<image.n_rows<<")!"
+    <<std::endl;
+  throw std::invalid_argument(oss.str());
+ }
+
+ kernel::GaussianKernel gaussianKernel = kernel::GaussianKernel(1.5);
+ arma::mat gaussianWindow(11, 11);
+ for ( int row = 0; row < 11; row++)
+ {
+  for ( int column = 0; column<11; column++)
+  {
+   gaussianWindow(row, column) = gaussianKernel.Evaluate(arma::vec(row - 5), arma::vec(column - 5));
```

Comment:
I think this is causing the error. `row - 5` would be < 0 and this would go into an infinite loop as there is no terminating condition or base case. Adding a check here should fix it.

Reply:
@kartikdutt18 Yes! That was the exact problem. Now on building and running the `./bin/mlpack_test -t CVTest` command, it says no errors detected.

Reply:
I am sorry it took so much time to reply. I decided to update from Ubuntu 18.04 to 20.04 and it took a lot of time.

Reply:
Awesome, There are also some style fixes that need to be applied. You could take a look at the logs [here](http://ci.mlpack.org/job/pull-requests%20mlpack%20style%20checks/4944/).

Reply:
No worries at all, Thanks again for taking the time to do this.